### ISSUE TITLE
🐛 Bug Report: Error: Missing required config value at 'app.https.certificate.cert' in 'app-config.yaml'

### ISSUE THREAD

**Issue by adminy** [backstage/backstage#28011]
### 📜 Description

I have a proxy (load balancer) sitting in front of the app component which does TLS termination.

so `app.baseUrl` should start with `https://` despite it doing http server. I thought that if `app.https.certificate` is not set, then it automatically assumes its http. However instead its just throwing the fact that its missing this cert.

### 👍 Expected behavior

if `app.https.certificate` is not set, don't expect it, just because `baseUrl` says its https.

### 👎 Actual Behavior with Screenshots

`baseUrl` dictates whether `app.https.certificate` should be set or not.

### 👟 Reproduction steps

set up demo but with `app.baseUrl` as a https link.

### 🖥️ Your Environment

OS:   Linux 6.12.1 - linux/x64
node: v20.16.0
yarn: 1.22.22
cli:  0.29.3-next.1 (installed)
backstage:  1.34.0-next.1

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by camilaibs** [comment]
Hi @adminy 👋🏻 , are you using `yarn start` in production to run the application? This configuration is only required for local development, which is why we are asking...

**Comment by mingfang** [comment]
I have the same setup for dev.
My load balancer does SSL termination.
Please make `app.https.certificate` optional

**Comment by freben** [comment]
@mingfang @adminy So you DO have a base URL with https in local dev mode (as opposed to production)? Are you doing this to be able to test out the proxy locally as well, or what's the use case? Normally we just have an `app-config.yaml` that has a http base url instead, which gets used for local dev.

**Comment by karthikjayaraman80** [comment]
I have the same problem. I use Github Codespaces and the base url is a https url in dev. 
yarn dev errors out with the above error.
Is there a workaround?

**Comment by freben** [comment]
I don't think so. We're open for contributions for this though.

**Comment by muja** [comment]
Same error here using OpenShift DevSpaces which forwards the development port. If we use `http://localhost:3003` then we get `Invalid Host header` error page, if we use the base url then we get this error...

Is there any possibility to overwrite the webpack config? I couldn't find anything on that, that would be a workaround..

**Comment by muja** [comment]
The workaround in our case was to overwrite the webpack config. I couldn't find a straight forward way for this, instead we had to resort to this hack:

```bash
sed -Ei "s/allowedHosts:.*/allowedHosts: 'all',/" node_modules/@backstage/cli/dist/cjs/index-nl4fBJN1.cjs.js
```

Is there no other way?

**Comment by camilaibs** [comment]
We could add some configuration for this, suggestions on this are welcome 🙏🏻

**Comment by adminy** [comment]
Yep, I wasn't packaging it in prod, just running `yarn start` which is why I got to see this. Thank you @camilaibs , @freben

**Comment by muja** [comment]
While I appreciate that your use case was solved, this issue is still very valid. Is it possible to reopen?
